pkg/operator/webhooksupportabilitycontroller/webhook_supportability_controller.go

```diff
@@ -59,6 +66,18 @@ func (c *webhookSupportabilityController) sync(ctx context.Context, controllerCo
 		return nil
 	}
 
+	// do nothing while an upgrade is in progress
+	clusterVersion, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		return err
+	}
+	desired := clusterVersion.Status.Desired.Version
+	history := clusterVersion.Status.History
+	// upgrade is in progress if there is no history, or the latest history entry matches the desired version and is not completed
+	if len(history) == 0 || history[0].Version != desired || history[0].State != configv1.CompletedUpdate {
+		return nil
```

Comment:
If we return `nil` here, does something go through and actually clear any of these `Degraded` cases from the status?  Because we could also have situations like:

1. Cluster is 4.11.0.
2. Broke webhook installed.
3. Controller goes degraded.
4. Admin requests an update to 4.11.1.
5. Controller notices it's updating, and clears these Degraded webhook health-checks.
6. Update to 4.11.1 completes.
7. Controller notices it's no longer updating, and resets the Degraded webhook health-checks.

Reply:
I think blocking 4.11 -> 4.11 and similar may be too strict, but we can always relax that later if I convince people.

/hold cancel